runtime/shared_common/shrinit.cpp

```diff
@@ -3509,6 +3525,19 @@ j9shr_init(J9JavaVM *vm, UDATA loadFlags, UDATA* nonfatal)
 		config->jclURLCache = NULL;
 		config->jclURLHashTable = NULL;
 		config->jclUTF8HashTable = NULL;
+		config->romToRamHashTable = hashTableNew(OMRPORT_FROM_J9PORT(vm->portLibrary),
```

Comment:
The hashtable and mutext should be created only when `J9_ARE_ALL_BITS_SET(config->runtimeFlags, J9SHR_RUNTIMEFLAG_ENABLE_CACHE_NON_BOOT_CLASSES)` is TRUE.

Also it is better to create the mutex before the hashtable. Access to the mutex and hashtable should be protected by `if (NULL != config->romToRamHashTable)`.